src/internal/ans.ts

```diff
@@ -171,5 +172,207 @@ export async function registerName(args: RegisterNameParameters): Promise<InputS
+        new U8(expiration.policy === "subdomain:follow-domain" ? 1 : 0),
+        new Bool(!!transferable),
+        new MoveOption(targetAddress ? AccountAddress.fromRelaxed(targetAddress) : null),
+        new MoveOption(toAddress ? AccountAddress.fromRelaxed(toAddress) : null),
+      ],
+    },
+    options,
+  });
+
+  return transaction as InputSingleSignerTransaction;
+}
+
+export async function getExpiration(args: { aptosConfig: AptosConfig; name: string }): Promise<number | undefined> {
+  const { aptosConfig, name } = args;
+  const routerAddress = getRouterAddress(aptosConfig);
+  const { domainName, subdomainName } = isValidANSName(name);
+
+  try {
+    const res = await view({
+      aptosConfig,
+      payload: {
+        function: `${routerAddress}::router::get_expiration`,
+        functionArguments: [domainName, Option(subdomainName)],
+      },
+    });
+
+    return res[0] as number;
+  } catch (e) {
+    return undefined;
+  }
+}
+
+export async function getPrimaryName(args: {
+  aptosConfig: AptosConfig;
+  address: AccountAddressInput;
+}): Promise<string | undefined> {
+  const { aptosConfig, address } = args;
+  const routerAddress = getRouterAddress(aptosConfig);
+
+  const res = await view({
+    aptosConfig,
+    payload: {
+      function: `${routerAddress}::router::get_primary_name`,
+      functionArguments: [AccountAddress.fromRelaxed(address).toString()],
+    },
+  });
+
+  const domainName = unwrapOption<MoveAddressType>(res[1]);
+  const subdomainName = unwrapOption<MoveAddressType>(res[0]);
+
+  if (!domainName) return undefined;
+
+  return [subdomainName, domainName].filter(Boolean).join(".");
+}
+
+export async function setPrimaryName(args: {
+  aptosConfig: AptosConfig;
+  sender: Account;
+  name: string;
+  options?: InputGenerateTransactionOptions;
+}): Promise<InputSingleSignerTransaction> {
+  const { aptosConfig, sender, name, options } = args;
+  const routerAddress = getRouterAddress(aptosConfig);
+  const { domainName, subdomainName } = isValidANSName(name);
+
+  const transaction = await generateTransaction({
+    aptosConfig,
+    sender: sender.accountAddress.toString(),
+    data: {
+      function: `${routerAddress}::router::set_primary_name`,
+      functionArguments: [
+        new MoveString(domainName),
+        new MoveOption(subdomainName ? new MoveString(subdomainName) : null),
+      ],
+    },
+    options,
+  });
+
+  return transaction as InputSingleSignerTransaction;
+}
+
+export async function getTargetAddress(args: {
+  aptosConfig: AptosConfig;
+  name: string;
+}): Promise<MoveAddressType | undefined> {
+  const { aptosConfig, name } = args;
+  const routerAddress = getRouterAddress(aptosConfig);
+  const { domainName, subdomainName } = isValidANSName(name);
+
+  const res = await view({
+    aptosConfig,
+    payload: {
+      function: `${routerAddress}::router::get_target_addr`,
+      functionArguments: [domainName, Option(subdomainName)],
```

Comment:
This one is a bit confusing. The `view` is returning an empty vector. I created [helper functions for this file](https://github.com/aptos-labs/aptos-ts-sdk/blob/d1a146761dac0b4442664e128ebadf2f8a6beeaa/src/internal/ans.ts#L84-L88) to do the unwrapping. Would it help if I changed the naming?

Reply:
ohh I see what you mean. @gregnazario have we introduced something to help with this? [helper functions for this file](https://github.com/aptos-labs/aptos-ts-sdk/blob/d1a146761dac0b4442664e128ebadf2f8a6beeaa/src/internal/ans.ts#L84-L88)

Reply:
View functions... no.  I can add some helpers / a remote ABI, but keeping in mind that it will need an entirely different input path than the current API (which is a real pain tbh).  I'm trying to get a different BCS update in for the next release on view functions on the API side.

Let's go forward what's here, and I can do a remote ABI refactor for this.

Reply:
should we at least have the SDK holds these functions and the view function to use it?

Reply:
for sure.